apps/gitness/src/pages/execution/execution-details.tsx

```diff
@@ -9,14 +9,30 @@ import {
   StageExecution,
   ContactCard,
   convertExecutionToTree,
-  StageProps
+  StageProps,
+  getStepId,
+  parseStageStepId
 } from '@harnessio/playground'
 import { PathParams } from '../../RouteDefinitions'
 import { useGetRepoRef } from '../../framework/hooks/useGetRepoPath'
 import { ExecutionState } from '../../types'
-import { getDuration, timeAgoFromEpochTime } from '../pipeline-edit/utils/time-utils'
+import { getDuration, timeAgoFromEpochTime, formatDuration } from '../pipeline-edit/utils/time-utils'
+import useSpaceSSE from '../../framework/hooks/useSpaceSSE'
+import { useGetSpaceURLParam } from '../../framework/hooks/useGetSpaceParam'
+
+/**
+ * Ensure this should come from @harnessio/code-service-client instead
+ */
+enum ExecutionEvent {
+  EXECUTION_UPDATED = 'execution_updated',
+  EXECUTION_COMPLETED = 'execution_completed',
+  EXECUTION_CANCELED = 'execution_canceled',
+  EXECUTION_RUNNING = 'execution_running'
+}
 
 const ExecutionLogs: React.FC = () => {
+  const navigate = useNavigate()
+  const space = useGetSpaceURLParam() ?? ''
```

Comment:
quick question: if space returns empty by chance , how will the page respond?

Reply:
It will just render empty page with no execution data, since the `/events` api itself gives 404 due to missing space path param.